From 3af72bb83b5b54be560da873ecd5fbaa6bd1999e Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Sun, 13 May 2018 18:03:35 +0300
Subject: [PATCH] sfp: failover to 1000basex

In case module does not provide line encoding, default to 1000basex 8 / 10
bit encoding.

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 drivers/net/phy/sfp-bus.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/phy/sfp-bus.c b/drivers/net/phy/sfp-bus.c
index 7bb6605..9b04dd4 100644
--- a/drivers/net/phy/sfp-bus.c
+++ b/drivers/net/phy/sfp-bus.c
@@ -108,7 +108,7 @@ phy_interface_t sfp_parse_interface(struct sfp_bus *bus,
 		break;
 
 	default:
-		iface = PHY_INTERFACE_MODE_NA;
+		iface = PHY_INTERFACE_MODE_1000BASEX;
 		dev_err(bus->sfp_dev,
 			"SFP module encoding does not support 8b10b nor 64b66b\n");
 		break;
-- 
1.7.9.5

